pkg/services/searchV2/bluge.go

```diff
@@ -65,12 +68,18 @@ func initIndex(dashboards []dashboard, logger log.Logger) (*bluge.Reader, *bluge
 		folderUID := folderIdLookup[dash.folderID]
 		location := folderUID
 		doc := getNonFolderDashboardDoc(dash, location)
+		if err := extendDoc(dash.uid, doc); err != nil {
+			return nil, nil, err
+		}
 		batch.Insert(doc)
 
 		// Index each panel in dashboard.
 		location += "/" + dash.uid
 		docs := getDashboardPanelDocs(dash, location)
 		for _, panelDoc := range docs {
+			if err := extendDoc(dash.uid, doc); err != nil {
```

Comment:
panelDoc should be here. And most probably panel unique UID? Maybe we can just omit extending panel at this point at all?